pkg/config/control_plane.go

```diff
@@ -186,9 +186,15 @@ type ControlPlaneCache struct {
 }
 
 type ControlPlaneInsightCollector struct {
-	Schedule          string `json:"schedule"`
-	RetryTime         int    `json:"retryTime"`
-	RetryIntervalHour int    `json:"retryIntervalHour"`
+	Schedule          string              `json:"schedule"`
+	RetryTime         int                 `json:"retryTime"`
+	RetryIntervalHour int                 `json:"retryIntervalHour"`
+	FeatureFlags      insightFeatureFlags `json:"featureFlags"`
+}
+
+type insightFeatureFlags struct {
```

Comment:
nit: How about `InsightCollectorDisabledMetrics` name? And by default, all metrics will be enabled.

``` golang
type InsightCollectorDisabledMetrics struct {
   DeploymentFrequency bool
   ChangeFailureRate bool
}
```

Reply:
I think it's good too 👍 
I don't have a particular opinion about which is better. So, I'll fix it